webapp/views.py

```diff
@@ -393,6 +394,44 @@ def get_notice(notice_id, **kwargs):
             404,
         )
 
+    releases = {
+        release.codename: release.version
+        for release in db_session.query(Release).all()
+    }
+
+    package_descriptions = {}
+    package_versions = {}
+    release_packages = SortedDict()
+
+    if notice.release_packages:
+        for codename, pkgs in notice.release_packages.items():
+            release_version = releases[codename]
+            release_packages[release_version] = {}
+            for package in pkgs:
+                if not package.get("is_visible", True):
+                    continue
+
+                name = package["name"]
+                if not package["is_source"]:
+                    release_packages[release_version][name] = package
+                    continue
+
+                if notice.notice_type == "LSN":
+                    if name not in package_descriptions:
+                        package_versions[name] = []
+
+                    package_versions[name].append(package["version"])
+                    versions = ", >= ".join(package_versions[name])
+                    description = (
+                        f"{package['description']} - " f"(>= {versions})"
+                    )
+                    package_descriptions[name] = description
+
+                    continue
+
+                if name not in package_descriptions:
+                    package_descriptions[name] = package["description"]
+
```

Comment:
So I'm confused. Now that I've got to the end of this new block of code it looks like you're populating `package_descriptions` and `release_packages` and then not doing anything with them - did I miss it? It looks like all this work just builds up some objects you then don't use.

I think I need you to explain this to me :)